### PR TITLE
Updating attendance records

### PR DESCRIPTION
**Pull request by dustinbrownman**

### CODE REVIEW

File: app/controllers/attendance_record_amendments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    @attendance_record_amendment = AttendanceRecordAmendment.new
+  end
+
+  def create
+    @attendance_record_amendment = AttendanceRecordAmendment.new(attendance_record_amendment_params)
+    if @attendance_record_amendment.save
+      student = Student.find(params[:attendance_record_amendment][:student_id])
+      redirect_to new_attendance_record_amendment_path, notice: "#{student.name}'s attendance record has been amended."
+    else
+      render :new, alert: 'Something went wrong. Please try again.'
```
Comment (michaelrkn) [inline app/controllers/attendance_record_amendments_controller.rb]: please surface the error to the user.

File: app/models/attendance_record_amendment.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    else
+      attendance_record = AttendanceRecord.find_or_initialize_by(student_id: @student_id, date: @date)
+      attendance_record.tardy = @status == "On time" ? false : true
+      attendance_record.save
+    end
+  end
+
+  def destroy_attendance_record
+    attendance_record = AttendanceRecord.find_by(student_id: @student_id, date: @date)
+    attendance_record.destroy unless attendance_record.nil?
```
Comment (michaelrkn) [inline app/models/attendance_record_amendment.rb]: `AttendanceRecord.find_by(student_id: @student_id, date: @date).try(:destroy)`

**Comment by michaelrkn** [comment]
closes #89 and #90.

**Comment by dustinbrownman** [comment]
I'm working on making these changes. Just having a heck of a time figuring
out the right way to handle errors.

**Comment by dustinbrownman** [comment]
This should be do it, but I'd like your (@michaelrkn) feedback first. Thanks!

File: app/controllers/attendance_record_amendments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  def new
+    @attendance_record_amendment = AttendanceRecordAmendment.new
+  end
+
+  def create
+    @attendance_record_amendment = AttendanceRecordAmendment.new(attendance_record_amendment_params)
+    if @attendance_record_amendment.save
+      student = Student.find(params[:attendance_record_amendment][:student_id])
+      redirect_to new_attendance_record_amendment_path, notice: "#{student.name}'s attendance record has been amended."
```
Comment (michaelrkn) [inline app/controllers/attendance_record_amendments_controller.rb]: `attendence_record_amendment.student.name`

**Pull request merged**